core/payment/src/dao/agreement.rs

```diff
@@ -128,7 +124,6 @@ pub fn increase_amount_paid(
     amount: &BigDecimalField,
     conn: &ConnType,
 ) -> DbResult<()> {
-    assert!(amount > &BigDecimal::zero().into()); // TODO: Remove when payment service is production-ready.
```

Comment:
Called from:
- requestor => notify_payment() 
- provider => verify_payment()

There are no checks if agreement payments are 0, should they be added? if yes, requestor, provider or both

Reply:
Yep, both `notify_payment` and `verify_payment` should check if amount > 0.

Reply:
added these checks :)